File: java/tesla-core/src/main/java/com/expedia/tesla/serialization/TeslaDeserializationException.java
/*******************************************************************************
 * Copyright (c) 2012, Expedia, Inc.
 *
 * TeslaDeserializationException.java
 *
 * Tesla Deserialization Exception.
 ******************************************************************************/

package com.expedia.tesla.serialization;

import java.io.IOException;

public class TeslaDeserializationException extends IOException {
	private static final long serialVersionUID = 8460203303982149351L;

	public TeslaDeserializationException(String message) {
		super(message);
	}

	public TeslaDeserializationException(String message, Throwable cause) {
		super(message, cause);
	}
}

File: csharp/TeslaDeserializationException.cs
﻿/*******************************************************************************
 * Copyright (c) 2014 Expedia Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *******************************************************************************/
namespace Expedia.Tesla.Serialization
{
    using System;

    /// <summary>
    /// Tesla Deserialization Exception.
    /// </summary>
    public class TeslaDeserializationException : TeslaException
    {
        /// <summary>
        /// Initializes a new instance of the TeslaDeserializationException class.
        /// </summary>
        /// <param name="msg">Error message</param>
        public TeslaDeserializationException(string msg)
            : base(msg)
        {
        }

        /// <summary>
        /// Initializes a new instance of the TeslaDeserializationException class.
        /// </summary>
        /// <param name="msg">Error message</param>
        /// <param name="innerException">The exception that is the cause of the current exception</param>
        public TeslaDeserializationException(string msg, Exception innerException)
            : base(msg, innerException)
        {
        }
    }
}